src/Shared/Microsoft.DotNet.Scaffolding.ComponentModel/CommandInfo.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.DotNet.Scaffolding.ComponentModel
+{
+    /// <summary>
+    /// What we want components to initialize and return (to stdout serialized as json) when 'get-commands' is invoked.
+    /// Should be serialized as 'List<CommandInfo>'
+    /// </summary>
+    public class CommandInfo
+    {
+        public string Name { get; set; } = default!;
```

Comment:
not sure how else to make the properties be required and be nullable friendly. how do i go about making them required?

Reply:
Using the [`required` modifier](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/required).  That's how we deal with a lot of nullable types in components in Blazor where it will be injected, e.g.:

https://github.com/dotnet/aspire/blob/f762d505df5947b8b9a7ce145a7ebc548616a521/src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs#L21

But how that will play with deserialization I'm not entirely sure. I just get a little antsy whenever I see the `!` as there's often, but not always, a way to avoid it.

Reply:
hmm, we can use required modifier and I'll write some unit tests for deserialization. That would give us a good picture of the expected behavior.

Reply:
added tests, more unit tests coming in the next PR.